spatialprofilingtoolbox/apiserver/app/validation.py

```diff
@@ -39,11 +44,10 @@ def valid_single_or_composite_identifier(identifier) -> str:
     abbreviation = abbreviate_string(identifier)
     raise ValueError(f'Channel name or phenotype identifier invalid: {abbreviation}')
 
-async def valid_phenotype1(phenotype1: str=Query(min_length=1)) -> str:
-    return valid_single_or_composite_identifier(phenotype1)
 
-async def valid_phenotype2(phenotype2: str=Query(min_length=1)) -> str:
-    return valid_single_or_composite_identifier(phenotype2)
+async def valid_phenotype(phenotype: str = Query(min_length=1)) -> str:
+    return valid_single_or_composite_identifier(phenotype)
+
```

Comment:
Unfortunately, FastAPI seems to pick up the attribute names verbatim and use them as the query parameter names, so I don't think that replacing `phenotype1` and `phenotype2` with `phenotype` in the `ValidPhenotype` works. I am getting errors about "phenotype" field missing, when "phenotype1" and "phenotype2" are supplied, even though they are called `phenotype1` and `phenotype2` in the call signature for `request_phenotype_proximity_computation`.
The workaround was to created separate types with the respective names (`ValidPhenotype1` and `ValidPhenotype2`).

Reply:
FastAPI more like SlowAndOvercomplicatedAPI